device/ethernet.yaml

```diff
@@ -1,58 +1,78 @@
-        ipv4_addresses:
-          description: |-
-            List of IPv4 addresses and their gateways. 
-          type: array
-          items:
-            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4' 
-        ipv6_addresses:
-          description: |-
-            List of global IPv6 addresses and their gateways.
-            The Link Local IPv6 address will be automatically generated.
-          type: array
-          items:
-            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
-      
\ No newline at end of file
+components:
+  schemas:
+    Device.EthernetBase:
+      x-include: 
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      description: >-
+        Base Ethernet interface.
+      type: object
+      required: [mac]
+      properties:
+        mac:
+          description: >-
+            Media Access Control address.
+          type: string
+          format: mac
+        mtu:
+          description: >-
+            Maximum Transmission Unit.
+          type: integer
+          minimum: 0
+          maximum: 65535
+          default: 1500
+        vlans:
+          description: >-
+            List of VLANs
+          type: array
+          items:
+            $ref: './vlan.yaml#/components/schemas/Device.Vlan'
+
+    Device.Ethernet:
+      x-include: 
+      - '#/components/schemas/Device.EthernetBase'
+      description: >-
+        An Ethernet interface with IPv4 and IPv6 addresses.
+      type: object
+      required: [connection]
+      properties:
+        connection: 
+          description: |-
+            Device connection to physical, LAG or another device.
+          $ref : '#/components/schemas/Device.Connection'
+        ipv4_addresses:
+          description: |-
+            List of IPv4 addresses and their gateways. 
+          type: array
+          items:
+            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4' 
+        ipv6_addresses:
+          description: |-
+            List of global IPv6 addresses and their gateways.
+            The Link Local IPv6 address will be automatically generated.
+          type: array
+          items:
+            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
+      
+    Device.Connection:
+      description: Device connection to physical port, LAG port or another device.
+      type: object
+      properties:
+        choice:
+          description: port_name, lag_name or device_name
+          type: string
+          enum: [port_name, lag_name, device_name]
+        port_name:
+          description: Name of the port the device is connected.
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+        lag_name:
+          description: Name of the Lag the device is connected.
+          type: string
+          x-constraint:
+          - '/components/schemas/Lag/properties/name'
+        device_name:
+          description: Connected/Emulated on top of another device

```

Comment:
nit - Name of the device that acts as the parent of this device.  ( i'm not sure of this, we should solicit some more feedback on exactly what this description should be, it would be good to get feedback from @apratimmukherjee )